scripts/verify-correctness.js

```diff
@@ -74,6 +87,17 @@ module.exports = async callback => {
         // If the last equation holds, the inverse trade must be profitable as well
       })
     )
+
+    // 5. verify that no bracket-trader offers profitable orders
+    await Promise.all(
+      bracketTraderAddresses.map(async bracketTrader => {
+        const relevantOrders = auctionElementsDecoded.filter(order => order.user.toLowerCase() == bracketTrader)
+        relevantOrders.forEach(async order =>
+          assert.equal(await checkNoProfitableOffer(order, BatchExchange, globalPriceStorage), true)
+        )
+      })
+    )
+
     callback()
```

Comment:
I think this block might not be working as intended: lines 94 to 97 define the mapping function, but the block has no return value. So the result of the mapping is a vector of `undefined` which is instantly resolved to itself by `Promise.all`. Then the function `callback()` is called in line 101, before the checks were completed. At this point I don't know what happens to the code that is being executed in the `forEach`.

Reply:
Is there a way to make js fail if `undefined` is used at any point in a computation in the code? This would have helped in many other occasions.

Reply:
Unfortunately, I am not aware of such an option. But yeah, this would have helped to spot many things. If we would have used typescript, this also would not have happened.